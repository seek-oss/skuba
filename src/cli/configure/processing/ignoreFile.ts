export const mergeWithIgnoreFile = (templateFile: string) => (
  inputFile?: string,
) => {
  if (typeof inputFile === 'undefined') {
    return `${templateFile}`;
  }

  if (inputFile.includes('# managed by skuba')) {
    const replacedSkubaSection = inputFile.replace(
      /# managed by skuba[\s\S]*# end managed by skuba/,
      templateFile,
    );

    return `${replacedSkubaSection}`;
  }

  const outputFile = [templateFile.trim(), inputFile.trim()]
    .join('\n\n')
    .trim();

  return `${outputFile}\n`;
};
