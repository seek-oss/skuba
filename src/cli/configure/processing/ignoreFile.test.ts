import { mergeWithIgnoreFile } from './ignoreFile';

describe('mergeWithIgnoreFile', () => {
  const baseTemplate =
    '# managed by skuba\nnode_modules\n# end managed by skuba';
  const updatedBaseTemplate =
    '# managed by skuba\nnode_modules\n.DS_Store\n# end managed by skuba';

  const cases = [
    ['empty provided', baseTemplate, ''],

    ['provided with no managed section', baseTemplate, '.DS_Store\n'],
    [
      'provided with outdated managed section',
      updatedBaseTemplate,
      `${baseTemplate}\n`,
    ],
    ['identical', baseTemplate, baseTemplate],
    [
      'provided with managed section and additional lines',
      baseTemplate,
      `.idea\n.vscode\n\n${baseTemplate}\n\n.DS_Store\nnode_modules\n`,
    ],
    [
      'provided with outdated managed section and additional lines',
      updatedBaseTemplate,
      `.idea\n.vscode\n\n${baseTemplate}\n\n.DS_Store\nnode_modules\n`,
    ],
  ] as const;

  test.each(cases)('%s', (_, base, provided) =>
    expect(mergeWithIgnoreFile(base)(provided)).toMatchSnapshot(),
  );
});
