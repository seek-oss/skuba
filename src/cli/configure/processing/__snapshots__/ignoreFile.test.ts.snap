// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mergeWithIgnoreFile empty provided 1`] = `
"# managed by skuba
node_modules
# end managed by skuba
"
`;

exports[`mergeWithIgnoreFile identical 1`] = `
"# managed by skuba
node_modules
# end managed by skuba"
`;

exports[`mergeWithIgnoreFile provided with managed section and additional lines 1`] = `
".idea
.vscode

# managed by skuba
node_modules
# end managed by skuba

.DS_Store
node_modules
"
`;

exports[`mergeWithIgnoreFile provided with no managed section 1`] = `
"# managed by skuba
node_modules
# end managed by skuba

.DS_Store
"
`;

exports[`mergeWithIgnoreFile provided with outdated managed section 1`] = `
"# managed by skuba
node_modules
.DS_Store
# end managed by skuba
"
`;

exports[`mergeWithIgnoreFile provided with outdated managed section and additional lines 1`] = `
".idea
.vscode

# managed by skuba
node_modules
.DS_Store
# end managed by skuba

.DS_Store
node_modules
"
`;
